python/ray/data/tests/test_dataset_parquet.py

```diff
@@ -31,6 +31,17 @@
 from pytest_lazyfixture import lazy_fixture
 
 
+def check_num_computed(ds, expected, streaming_expected) -> None:
+    # When streaming executor is on, the _num_computed() is affected only
+    # by the ds.schema() which will still partial read the blocks, but will
+    # not affected by operations like take() as it's executed via streaming
+    # executor.
```

Comment:
Ideally we'd check the number of computed blocks using `ds.stats()`, but I guess this can be nondeterministic in streaming mode.

Actually, are you sure this current check isn't flaky (doesn't schema() also trigger streaming execution)? Maybe we can assert a lower bound / upper bound instead of an exact number.

Reply:
The schema() run this partial execution: https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/plan.py#L395
So even if it's streaming, `_num_computed()` can be non-zero (i.e. if such partial execution is never hit in streaming, we can simply assert 0 here).